src/app/business/routes/algo/sagas/index.js

```diff
@@ -8,14 +8,91 @@ import {saveAs} from 'file-saver';
 import cookie from 'cookie-parse';
 
 import actions, {actionTypes} from '../actions';
-import {fetchListApi, fetchItemApi} from '../api';
 import {
-fetchListSaga, fetchPersistentSaga, fetchItemSaga, setOrderSaga,
+    fetchListApi, fetchStandardAlgoApi, fetchCompositeAlgoApi, fetchAggregateAlgoApi,
+} from '../api';
+import {
+fetchItemSaga, setOrderSaga,
 } from '../../../common/sagas';
 import {fetchRaw} from '../../../../entities/fetchEntities';
 import {getItem} from '../../../common/selector';
 import {signOut} from '../../../user/actions';
 
+const fetchItemApiByType = {
+    composite: fetchCompositeAlgoApi,
+    aggregate: fetchAggregateAlgoApi,
+    standard: fetchStandardAlgoApi,
+};
+
+const withAlgoType = (list, type) => list.map(group => group.map(algo => ({...algo, type})));
+
+
+const fetchListSaga = (actions, fetchListApi) => function* fetchList({payload}) {
+    const [resStandardAlgos, resCompositeAlgos, resAggregateAlgos] = yield call(fetchListApi, payload);
+    const {error: errorStandardAlgos, status: statusStandardAlgos, list: listStandardAlgos} = resStandardAlgos;
+    const {error: errorCompositeAlgos, status: statusCompositeAlgos, list: listCompositeAlgos} = resCompositeAlgos;
+    const {error: errorAggregateAlgos, status: statusAggregateAlgos, list: listAggregateAlgos} = resAggregateAlgos;
+
+    let list;
+
+    if (errorStandardAlgos) {
+        console.error(errorStandardAlgos, statusStandardAlgos);
+        yield put(actions.list.failure(errorStandardAlgos));
+    }
+    else if (errorCompositeAlgos) {
+        console.error(errorCompositeAlgos, statusCompositeAlgos);
+        yield put(actions.list.failure(errorCompositeAlgos));
+    }
+    else if (errorAggregateAlgos) {
+        console.error(errorAggregateAlgos, statusAggregateAlgos);
+        yield put(actions.list.failure(errorAggregateAlgos));
+    }
+    else {
+        list = [].concat(
+            withAlgoType(listStandardAlgos, 'standard'),
+            withAlgoType(listCompositeAlgos, 'composite'),
+            withAlgoType(listAggregateAlgos, 'aggregate'),
+        );
```

Comment:
The goal here was to abstract completely the complexity of these 3 api endpoints. I thought of managing them independently but it would have required a lot more changes because of the structure changes and a lot of code duplication to accommodate the small changes needed in the common utils (selectors, filters, etc.).